pkg/runtime/wfengine/activity_test.go

```diff
@@ -30,10 +30,10 @@ import (
 func TestDedupeActivityInvocation(t *testing.T) {
 	// We do just enough to get the activity actor properly initialized
 	ctx := context.Background()
-	_, engine := startEngine(ctx, task.NewTaskRegistry())
+	_, engine := startEngine(ctx, t, task.NewTaskRegistry())
```

Comment:
every occurrence of `startEngine` seems to be preceded with a `context.Background` call ... Do you really need context to be passed in, can it just be init inside the `startEngine` function itself?

Reply:
Can do so but let's take a case like `TestEmptyWorkflow`: https://github.com/dapr/dapr/blob/3fb850d12c5306f1dd424cfa15b41d27f0c4e0e7/pkg/runtime/wfengine/wfengine_test.go#L107

There, we also need to pass `context` to `ScheduleNewOrchestration` and `WaitForOrchestrationCompletion`, and if I understand correctly, a Prod scenario will have same context being passed in there after starting Engine and then to schedule and wait for an orchestration. So, current test mimics almost similar scenario with same context being passed to them.
The tests would ideally also work with `context.TODO()` in all these places. But, I feel it is better to use a common context, which is being done currently.
WDYT?